drivers/gnss/gnss_quectel_lcx6g.c

```diff
@@ -0,0 +1,794 @@
+}
+
+static void quectel_lcx6g_init_pipe(const struct device *dev)
+{
+	const struct quectel_lcx6g_config *config = dev->config;
+	struct quectel_lcx6g_data *data = dev->data;
+
+	const struct modem_backend_uart_config uart_backend_config = {
+		.uart = config->uart,
+		.receive_buf = data->uart_backend_receive_buf,
+		.receive_buf_size = ARRAY_SIZE(data->uart_backend_receive_buf),
+		.transmit_buf = data->uart_backend_transmit_buf,
+		.transmit_buf_size = ARRAY_SIZE(data->uart_backend_transmit_buf),
+	};
+
+	data->uart_pipe = modem_backend_uart_init(&data->uart_backend, &uart_backend_config);
+}
+
+static int quectel_lcx6g_init_chat(const struct device *dev)
+{
+	struct quectel_lcx6g_data *data = dev->data;
+
+	const struct modem_chat_config chat_config = {
+		.user_data = data,
+		.receive_buf = data->chat_receive_buf,
+		.receive_buf_size = ARRAY_SIZE(data->chat_receive_buf),
+		.delimiter = data->chat_delimiter,
+		.delimiter_size = ARRAY_SIZE(data->chat_delimiter),
+		.filter = NULL,
+		.filter_size = 0,
+		.argv = data->chat_argv,
+		.argv_size = ARRAY_SIZE(data->chat_argv),
+		.unsol_matches = unsol_matches,
+		.unsol_matches_size = ARRAY_SIZE(unsol_matches),
+		.process_timeout = K_MSEC(2),
+	};
+
+	return modem_chat_init(&data->chat, &chat_config);
+}
+
+static void quectel_lcx6g_init_dynamic_script(const struct device *dev)
+{
+	struct quectel_lcx6g_data *data = dev->data;
+
+	data->dynamic_match.match = data->dynamic_match_buf;
+	data->dynamic_match.separators = data->dynamic_separators_buf;
+	data->dynamic_match.separators_size = sizeof(data->dynamic_separators_buf);
+	data->dynamic_match.wildcards = false;
+	data->dynamic_match.partial = false;
+
+	data->dynamic_script_chat.request = data->dynamic_request_buf;
+	data->dynamic_script_chat.response_matches = &data->dynamic_match;
+	data->dynamic_script_chat.response_matches_size = 1;
+	data->dynamic_script_chat.timeout = 0;
+
+	data->dynamic_script.name = "pair";
+	data->dynamic_script.script_chats = &data->dynamic_script_chat;
+	data->dynamic_script.script_chats_size = 1;
+	data->dynamic_script.abort_matches = NULL;
+	data->dynamic_script.abort_matches_size = 0;
+	data->dynamic_script.callback = NULL;
+	data->dynamic_script.timeout = 10;
+}
+
+static int quectel_lcx6g_init(const struct device *dev)
+{
+	int ret;
+
+	LOG_INF("Initializing Quectel LCX6G");
+	ret = quectel_lcx6g_init_nmea0183_match(dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	quectel_lcx6g_init_pipe(dev);
+
+	ret = quectel_lcx6g_init_chat(dev);
+	if (ret < 0) {
+		return ret;
+	}
+
+	quectel_lcx6g_init_dynamic_script(dev);
+
+#ifdef CONFIG_PM_DEVICE
+	pm_device_init_suspended(dev);
+#else
+	LOG_INF("Resuming Quectel LCX6G");
+	ret = quectel_lcx6g_resume(dev);
+	if (ret < 0) {
+		LOG_ERR("Failed to resume Quectel LCX6G");
+		return ret;
+	}
+#endif
+	LOG_INF("Quectel LCX6G initialized");
+	return 0;
+}
+
+#define LCX6G_INST_NAME(inst, name) \
+	_CONCAT(_CONCAT(_CONCAT(name, _), DT_DRV_COMPAT), inst)
```

Comment:
why is this needed?

Reply:
Because of these:
```
#define DT_DRV_COMPAT quectel_lc26g
DT_INST_FOREACH_STATUS_OKAY(LCX6G_DEVICE)
#undef DT_DRV_COMPAT

#define DT_DRV_COMPAT quectel_lc76g
DT_INST_FOREACH_STATUS_OKAY(LCX6G_DEVICE)
#undef DT_DRV_COMPAT

#define DT_DRV_COMPAT quectel_lc86g
DT_INST_FOREACH_STATUS_OKAY(LCX6G_DEVICE)
#undef DT_DRV_COMPAT
```

they could share the same instance, so the name needs to be unique to the specific revision.
```
data_quectel_lc26g0
data_quectel_lc26g1

data_quectel_lc76g0
```